File: src/models/cache_create_namespace_out.rs
// this file is @generated
use serde::{Deserialize, Serialize};

use super::eviction_policy::EvictionPolicy;

#[derive(Clone, Debug, Deserialize, Serialize)]
pub struct CacheCreateNamespaceOut {
    pub name: String,

    #[serde(skip_serializing_if = "Option::is_none")]
    pub max_storage_bytes: Option<u64>,

    pub eviction_policy: EvictionPolicy,

    pub created: jiff::Timestamp,

    pub updated: jiff::Timestamp,
}

impl CacheCreateNamespaceOut {
    pub fn new(
        name: String,
        eviction_policy: EvictionPolicy,
        created: jiff::Timestamp,
        updated: jiff::Timestamp,
    ) -> Self {
        Self {
            name,
            max_storage_bytes: None,
            eviction_policy,
            created,
            updated,
        }
    }

    pub fn with_max_storage_bytes(mut self, value: impl Into<Option<u64>>) -> Self {
        self.max_storage_bytes = value.into();
        self
    }
}

File: coyote/models/cache_create_namespace_out.py
# this file is @generated
import typing as t
from pydantic import Field
from datetime import datetime

from ..internal.base_model import BaseModel

from .eviction_policy import EvictionPolicy


class CacheCreateNamespaceOut(BaseModel):
    name: str

    max_storage_bytes: t.Optional[int] = Field(default=None, alias="max_storage_bytes")

    eviction_policy: EvictionPolicy = Field(alias="eviction_policy")

    created: datetime

    updated: datetime
